### DISCUSSION TITLE
Is there a way to run one process/pod per node using k8s executor?

### DISCUSSION THREAD

**Discussion by luisovallex** [nextflow-io/nextflow#2679] [Q&A] [answered]
I'm trying to run only one process per node on EC2 instances and using k8s as executor for a nextflow pipeline, but I've seen that even if I adjust the memory of the process on a node it will try to allocate two or more process in the instances that I'm using, is this possible to achieve?

**Comment by bentsherman** [comment, marked as answer]
Make sure you specify both a request and a limit for cpu and memory. if that doesn't work, I would look into using affinity or tolerations/taints to create a rule that would force one pod per node. Nextflow support for affinity was added recently and should be in the latest edge release. Tolerations/taints aren't supported yet but I plan to implement them soon.
